swift/swift5/Swift5.g4

```diff
@@ -0,0 +1,1372 @@
+	| [\u{40000}-\u{4FFFD}]
+	| [\u{50000}-\u{5FFFD}]
+	| [\u{60000}-\u{6FFFD}]
+	| [\u{70000}-\u{7FFFD}]
+	| [\u{80000}-\u{8FFFD}]
+	| [\u{90000}-\u{9FFFD}]
+	| [\u{A0000}-\u{AFFFD}]
+	| [\u{B0000}-\u{BFFFD}]
+	| [\u{C0000}-\u{CFFFD}]
+	| [\u{D0000}-\u{DFFFD}]
+	| [\u{E0000}-\u{EFFFD}];
+
+fragment Identifier_character:
+	[0-9]
+	| [\u0300-\u036F]
+	| [\u1DC0-\u1DFF]
+	| [\u20D0-\u20FF]
+	| [\uFE20-\uFE2F]
+	| Identifier_head;
+
+fragment Identifier_characters: Identifier_character+;
+
+fragment Implicit_parameter_name: '$' Decimal_digits;
+
+fragment Property_wrapper_projection: '$' Identifier_characters;
+
+// Keywords and Punctuation
+Keyword:
+	// Keywords used in declarations
+	'associatedtype'
+	| 'class'
+	| 'deinit'
+	| 'enum'
+	| 'extension'
+	| 'fileprivate'
+	| 'func'
+	| 'import'
+	| 'init'
+	| 'inout'
+	| 'internal'
+	| 'let'
+	| 'open'
+	| 'operator'
+	| 'private'
+	| 'protocol'
+	| 'public'
+	| 'rethrows'
+	| 'static'
+	| 'struct'
+	| 'subscript'
+	| 'typealias'
+	| 'var'
+	// Keywords used in statements
+	| 'break'
+	| 'case'
+	| 'continue'
+	| 'default'
+	| 'defer'
+	| 'do'
+	| 'else'
+	| 'fallthrough'
+	| 'for'
+	| 'guard'
+	| 'if'
+	| 'in'
+	| 'repeat'
+	| 'return'
+	| 'switch'
+	| 'where'
+	| 'while'
+	// Keywords used in expressions and types
+	| 'as'
+	| 'Any'
+	| 'catch'
+	| 'false'
+	| 'is'
+	| 'nil'
+	| 'super'
+	| 'self'
+	| 'Self'
+	| 'throw'
+	| 'throws'
+	| 'true'
+	| 'try'
+	// Keywords used in patterns
+	| UNDERSCORE
+	// Keywords that begin with a number sign (#)
+	| '#available'
+	| '#colorLiteral'
+	| '#column'
+	| '#else'
+	| '#elseif'
+	| '#endif'
+	| '#error'
+	| '#file'
+	| '#fileID'
+	| '#fileLiteral'
+	| '#filePath'
+	| '#function'
+	| '#if'
+	| '#imageLiteral'
+	| '#line'
+	| '#selector'
+	| '#sourceLocation'
+	| '#warning';
+
+// Operators
+
+// Assignment Operator
+assignment_operator: {SwiftSupport.isBinaryOp(_input)}? '=';
+
+DOT: '.';
+LCURLY: '{';
+LPAREN: '(';
+LBRACK: '[';
+RCURLY: '}';
+RPAREN: ')';
+RBRACK: ']';
+COMMA: ',';
+COLON: ':';
+SEMI: ';';
+LT: '<';
+GT: '>';
+UNDERSCORE: '_';
+BANG: '!';
+QUESTION: '?';
+AT: '@';
+AND: '&';
+SUB: '-';
+EQUAL: '=';
+OR: '|';
+DIV: '/';
+ADD: '+';
+MUL: '*';
+MOD: '%';
+CARET: '^';
+TILDE: '~';
+
+negate_prefix_operator: {SwiftSupport.isPrefixOp(_input)}? '-';
+
+compilation_condition_AND: {SwiftSupport.isOperator(_input,"&&")}? '&' '&';
+compilation_condition_OR: {SwiftSupport.isOperator(_input,"||")}? '|' '|';
+compilation_condition_GE: {SwiftSupport.isOperator(_input,">=")}? '>' '=';
+compilation_condition_L: {SwiftSupport.isOperator(_input,"<")}? '<';
+arrow_operator: {SwiftSupport.isOperator(_input,"->")}? '-' '>';
+range_operator: {SwiftSupport.isOperator(_input,"...")}? '.' '.' '.';
+same_type_equals: {SwiftSupport.isOperator(_input,"==")}? '=' '=';
+
+binary_operator: {SwiftSupport.isBinaryOp(_input)}? operator;
+
+prefix_operator: {SwiftSupport.isPrefixOp(_input)}? operator;
+
+postfix_operator: {SwiftSupport.isPostfixOp(_input)}? operator;
```

Comment:
I don't recommend using predicates at the beginning for lexer rules because it degrades the performance. Use predicates in the following way (semantics of `isPrefixOp` is also should be changed):

```g4
negate_prefix_operator: '-' {SwiftSupport.isPrefixOp(_input)}?;
```

See detail in the article [Improving the performance of an ANTLR parser](https://tomassetti.me/improving-the-performance-of-an-antlr-parser/).